webapi/src/main/scala/org/knora/webapi/util/date/CalendarDateUtilV2.scala

```diff
@@ -0,0 +1,543 @@
+/*
+ * Copyright © 2015-2018 the contributors (see Contributors.md).
+ *
+ * This file is part of Knora.
+ *
+ * Knora is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Knora is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with Knora.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.knora.webapi.util.date
+
+import java.util.Date
```

Comment:
Is the Java date class (GregorianCalendar) used anywhere in this code?

Reply:
No.

Reply:
`com.ibm.icu.util.GregorianCalendar` replaces `java.util.GregorianCalendar`.

Reply:
http://icu-project.org/apiref/icu4j/com/ibm/icu/util/GregorianCalendar.html

> ICU's replacement for java.util.GregorianCalendar.

Reply:
I'm using the ICU `GregorianCalendar` class instead of the one in the Java standard library, because the ICU class supports Julian Day Numbers.